### ISSUE TITLE
Feature request: add -x/--line-regexp (match whole line) option

### ISSUE THREAD

**Issue by okdana** [ggreer/the_silver_searcher#626]
I've already submitted a pull request for this (#625), but i thought i ought to create an issue as well so that people can find it more easily.

Summary: Requesting an `ag -x` option equivalent to `grep -x`, which forces the pattern to match only against whole lines

**Comment by ggreer** [comment]
I'll leave your issues open, but in the future, please don't create issues for each pull request. Over 200 people get emailed for each issue or PR, and searching the whole project isn't difficult.

**Comment by okdana** [comment]
Oh, i'm sorry. I'm used to needing a ticket to document every change, i don't have much experience with GitHub etiquette. Sorry again for the spam!

**Comment by ggreer** [comment]
Yeah, I figured it was an honest mistake. Don't feel bad about it.

**Comment by cup** [comment]
@ggreer any comment on the actual issue? `-x` is not just defined by GNU, but POSIX as well:

http://pubs.opengroup.org/onlinepubs/9699919799/utilities/grep.html

granted, Ag is not a POSIX tool, but its worth mentioning
